src/Analyser/MutatingScope.php

```diff
@@ -4843,7 +4844,35 @@ private static function generalizeType(Type $a, Type $b): Type
 			$resultTypes[] = TypeCombinator::union(...$integerRanges['b']);
 		}
 
-		return TypeCombinator::union(...$resultTypes, ...$otherTypes);
+		$accessoryTypes = [];
+		$aAccessoryTypes = TypeUtils::getAccessoryTypes($a);
+		$bAccessoryTypes = TypeUtils::getAccessoryTypes($b);
+		if (count($aAccessoryTypes) > 0) {
+			if (count($bAccessoryTypes) === 0) {
+				$accessoryTypes = $aAccessoryTypes;
+			} else {
+				$accessoryTypesToUnion = [];
+				foreach (array_merge($aAccessoryTypes, $bAccessoryTypes) as $accessoryType) {
```

Comment:
What doesn't look right here is that accessory type should be preserved only if it's on both sides of the generalization - A and B. That's what is acomplished by array_intersect_key in TypeCombinator.

Reply:
This line https://github.com/phpstan/phpstan-src/blob/bd57fc59f4737188ce750f607f42d85e6d4aec37/src/Type/TypeCombinator.php#L216

Reply:
I had this at some point and threw it away again :)
Do you think it should always work like that? E.g. also for the count conditionals where either a or b accessories are empty and the other one is fully taken over? That would most likely fix the ignore / test failures I guess. Not sure yet if it still fixes the linked issue, hut I can find out